### PR TITLE
Server Engine 2

### PR DESCRIPTION
**Pull request by notzippy**
Minimum changes for server engine to work

### CODE REVIEW

File: harness/harness.go [hunk 1]
```diff
@@ -47,10 +47,12 @@ type Harness struct {
 	proxy      *httputil.ReverseProxy
 }
 
-func renderError(w http.ResponseWriter, r *http.Request, err error) {
-	req, resp := revel.NewRequest(r), revel.NewResponse(w)
-	c := revel.NewController(req, resp)
-	c.RenderError(err).Apply(req, resp)
+func renderError(iw http.ResponseWriter, ir *http.Request, err error) {
+	context := revel.NewGOContext(nil)
```
Comment (brendensoares) [inline harness/harness.go]: why `GO` and why is this code calling a specific server engine context?

File: revel/run.go [hunk 2]
```diff
@@ -89,7 +89,12 @@ func parseRunArgs(args []string) *RunArgs {
 		// 1. revel run [import-path]
 		// 2. revel run [port]
 		// 3. revel run [run-mode]
-		if _, err := build.Import(args[0], "", build.FindOnly); err == nil {
+		_, err := build.Import(args[0], "", build.FindOnly)
+		if err != nil {
+			revel.WARN.Printf("Unable to run using an import path, assuming import path is working directory %s %s", args[0], err.Error())
```
Comment (brendensoares) [inline revel/run.go]: how can we test this?

File: revel/skeleton/app/init.go [hunk 3]
```diff
@@ -29,7 +29,6 @@ func init() {
 		revel.ActionInvoker,           // Invoke the action.
 	}
 
-
 	// register startup functions with OnAppStart
```
Comment (brendensoares) [inline revel/skeleton/app/init.go]: `Register`

File: harness/harness.go (hunk 1 above)
Comment (notzippy) [inline harness/harness.go]: This code need to run a specific engine to display any compile errors, if the compile succeeds then the application is launched and this turns into a proxy

File: revel/run.go (hunk 2 above)
Comment (notzippy) [inline revel/run.go]: This occurs when you do not specify the application to run. Currently the command silently fails and assumes that the working directory is what you want to launch. This typically can cause really misleading error messages down the road if that was not the case. With this catch at least you have some idea what revel did

**Pull request merged**